src/Frame/Frame.cc

```diff
@@ -1612,76 +1581,71 @@ bool Frame::HasSpectralConfig(const SpectralConfig& config) {
 // Region/Slicer Support (Frame manages image mutex)
 
 std::shared_ptr<casacore::LCRegion> Frame::GetImageRegion(
-    int file_id, std::shared_ptr<Region> region, const StokesSource& stokes_source, bool report_error) {
+    int file_id, std::shared_ptr<Region> region, int stokes_index, bool report_error) {
     // Return LCRegion formed by applying region params to image.
     // Returns nullptr if region outside image
-    return region->GetImageRegion(file_id, CoordinateSystem(stokes_source), ImageShape(stokes_source), stokes_source, report_error);
+    return region->GetImageRegion(file_id, CoordinateSystem(stokes_index), ImageShape(stokes_index), stokes_index, report_error);
 }
 
-bool Frame::GetImageRegion(int file_id, const AxisRange& z_range, int stokes, StokesRegion& stokes_region) {
-    if (!CheckZ(z_range.from) || !CheckZ(z_range.to) || !CheckStokes(stokes)) {
+bool Frame::GetImageRegion(int file_id, const AxisRange& z_range, int stokes_index, casacore::ImageRegion& image_region) {
+    if (!CheckZ(z_range.from) || !CheckZ(z_range.to) || !CheckStokes(stokes_index)) {
         return false;
     }
     try {
-        StokesSlicer stokes_slicer = GetImageSlicer(z_range, stokes);
-        stokes_region.stokes_source = stokes_slicer.stokes_source;
-        casacore::LCSlicer lcslicer(stokes_slicer.slicer);
+        casacore::Slicer slicer = GetImageSlicer(z_range, stokes_index);
+        casacore::LCSlicer lcslicer(slicer);
         casacore::ImageRegion this_region(lcslicer);
-        stokes_region.image_region = this_region;
+        image_region = this_region;
         return true;
     } catch (casacore::AipsError error) {
         spdlog::error("Error converting full region to file {}: {}", file_id, error.getMesg());
         return false;
     }
 }
 
-casacore::IPosition Frame::GetRegionShape(const StokesRegion& stokes_region) {
+casacore::IPosition Frame::GetRegionShape(const casacore::LattRegionHolder& region, int stokes_index) {
     // Returns image shape with a region applied
-    auto coord_sys = CoordinateSystem(stokes_region.stokes_source);
-    casacore::LatticeRegion lattice_region =
-        stokes_region.image_region.toLatticeRegion(*coord_sys.get(), ImageShape(stokes_region.stokes_source));
+    auto coord_sys = CoordinateSystem(stokes_index);
+    casacore::LatticeRegion lattice_region = region.toLatticeRegion(*coord_sys.get(), ImageShape(stokes_index));
     return lattice_region.shape();
 }
 
-bool Frame::GetRegionSubImage(const StokesRegion& stokes_region, casacore::SubImage<float>& sub_image) {
+bool Frame::GetRegionSubImage(const casacore::LattRegionHolder& region, int stokes_index, casacore::SubImage<float>& sub_image) {
```

Comment:
We should note that LattRegionHolder is the base class of ImageRegion, or just use ImageRegion (as in GetRegionData)